Source/RunActivity/Viewer3D/Cameras.cs

```diff
@@ -1885,48 +1899,202 @@ public override void Update(ElapsedTime elapsedTime)
 
             // Otherwise, let's check out every car and remember which is the first one close enough for next time.
             if (!trainClose)
             {
                 foreach (var car in train.Cars)
                 {
-                    if (WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < MaximumDistance)
+                    if (LastCheckCar != null && car == LastCheckCar && 
+                        WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < (SpecialPointFound ? MaximumSpecialPointDistance * 0.8f : MaximumDistance))
                     {
+                        trainClose = true;
+                        break;
+                    }
+                    else if (WorldLocation.GetDistance2D(car.WorldPosition.WorldLocation, cameraLocation).Length() < MaximumDistance)
+                        {
                         LastCheckCar = car;
                         trainClose = true;
                         break;
                     }
                 }
+                if (!trainClose)
+                    LastCheckCar = null;
+            }
+            var trySpecial = false;
+            DistanceRunM += elapsedTime.ClockSeconds * train.SpeedMpS;
+            if (Math.Abs(DistanceRunM) >= CheckIntervalM)
+            {
+                DistanceRunM = 0;
+                if (!SpecialPointFound && SpecialPoints && trainClose) trySpecial = true;
             }
-
             // Switch to new position.
-            if (!trainClose || (TrackCameraLocation == WorldLocation.None))
-            {
-                var tdb = trainForwards ? new Traveller(train.FrontTDBTraveller) : new Traveller(train.RearTDBTraveller, Traveller.TravellerDirection.Backward);
-                tdb.Move(MaximumDistance * 0.75f);
-                var newLocation = tdb.WorldLocation;
-                TrackCameraLocation = new WorldLocation(newLocation);
-                var directionForward = WorldLocation.GetDistance((trainForwards ? train.FirstCar : train.LastCar).WorldPosition.WorldLocation, newLocation);
-                if (Viewer.Random.Next(2) == 0)
+            if (!trainClose || (TrackCameraLocation == WorldLocation.None) || trySpecial)
+            {
+                SpecialPointFound = false;
+                bool platformFound = false;
+                Traveller tdb;
+                if (FirstUpdateLoop && SpecialPoints) 
+                    tdb = trainForwards ? new Traveller(train.RearTDBTraveller) : new Traveller(train.FrontTDBTraveller, Traveller.TravellerDirection.Backward);
+                 else
+                    tdb = trainForwards ? new Traveller(train.FrontTDBTraveller) : new Traveller(train.RearTDBTraveller, Traveller.TravellerDirection.Backward);
+                WorldLocation newLocation = WorldLocation.None;
```

Comment:
Where possible, it's preferable to use `var` for local variables.

Reply:
Done